soc/riscv/riscv-privilege/hpmicro/soc.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/irq.h>
+#include <hpm_common.h>
+#include <hpm_soc.h>
+#include <board.h>
+#include "hpm_pmp_drv.h"
+#include "hpm_pllctl_drv.h"
+#ifdef CONFIG_XIP
+#include "hpm_bootheader.h"
+#endif
+
+#ifdef CONFIG_XIP
+__attribute__ ((section(".nor_cfg_option"))) const uint32_t option[4] = {
```

Comment:
A comment explaining what this does and why it is needed would be nice.

Reply:
The rom boot of the chip will get the content of this field to initialize the spi flash in the rom.

Reply:
Ditto - explanation in this file, not in this convo